src/main/java/com/xebialabs/restito/semantics/Condition.java

```diff
@@ -200,6 +209,33 @@ public static Condition withHeader(final String key, final String value) {
     }
 
     /**
+     * With header present and contains all values.
+     * Header key is case insensitive. More information <a href="http://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive">here</a>.
+     */
+    public static Condition withMultiHeaderContains(final String key, final String... values) {
+        return new Condition(input -> {
+            final String[] actualHeaderValues = input.getMultiHeaders().get(key);
+            return Optional.ofNullable(actualHeaderValues)
+                           .filter(it -> Stream.of(values).allMatch(val -> Arrays.binarySearch(it, val) >= 0))
+                           .isPresent();
+        });
+    }
+
+    /**
+     * With header present and matches all values. Header should contain exact number of elements. Order does not matter.
+     * Header key is case insensitive. More information <a href="http://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive">here</a>.
+     */
+    public static Condition withMultiHeaderMatches(final String key, final String... values) {
```

Comment:
If you introduce `withMultiHeaderMatches`, please also create `withHeaderMatches`, for consistency.